### PR TITLE
Unpin url launcher (remake)

### LINKED ISSUE #111304: Un-pin url_launcher_android in flutter/flutter
PR with broken build tests: https://github.com/flutter/flutter/pull/111197/files#diff-9aeff568c87b1d587222c343afb84ef72fc07e0692d50ecb3de670e5606f9505R46

I believe it is this change: https://github.com/flutter/plugins/pull/6194. Note, something similar happened in https://github.com/flutter/flutter/issues/109397 with path_provider_android, and what happened was that the change was reverted upstream and a new version released.

I imagine we will eventually need to find a way to get the migrated versions of these plugins passing flutter/flutter CI.

### PR DESCRIPTION
**Pull request by gmackall**
More up to date version of https://github.com/flutter/flutter/pull/133786.

Fixes https://github.com/flutter/flutter/issues/111304

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gmackall** [comment]
@christopherfujino discussed internally regarding the google testing and got the advice to just try overriding. 

This PR was made the same way as the previous one, just more recently. Sorry for the review churn!

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by ditman** [comment]
@christopherfujino this one doesn't seem to contain the bump to `package:web`?

File: dev/a11y_assessments/pubspec.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ dependencies:
   characters: 1.3.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
   collection: 1.18.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
   material_color_utilities: 0.5.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  meta: 1.9.1 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
+  meta: 1.10.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
   vector_math: 2.1.4 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
   web: 0.1.4-beta # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
```
Comment (ditman) [inline dev/a11y_assessments/pubspec.yaml:16]: /cc @christopherfujino this one doesn't seem to be updating the `package:web` dep? Can we run another `flutter update-packages --force-upgrade` here?

Comment (ditman) [reply dev/a11y_assessments/pubspec.yaml:16]: (I can also update https://github.com/flutter/flutter/pull/135081 after this one lands, whatever you guys prefer!)

Comment (christopherfujino) [reply dev/a11y_assessments/pubspec.yaml:16]: as Gray has been trying to update this for weeks, I added autosubmit and got it merged. Can you update 135081? I'll approve now

Comment (ditman) [reply dev/a11y_assessments/pubspec.yaml:16]: @christopherfujino will do! ;)

**Pull request merged**